### PR TITLE
Requirements Cleanup

### PR DESCRIPTION
**Pull request by ahsan-ul-haq**
@clintonb I have updated the setup please take a look.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -25,6 +25,8 @@
     install_requires=[
         "django>=1.8",
         "django-model-utils==1.4.0",
+        "djangorestframework==3.2.3",
```
Comment (clintonb) [inline setup.py]: These should not be hard requirements. Packages need to be able to work with multiple versions: `djangorestframework>=3.2.0,<3.4.0` (I *think* 3.3.x should work.)

Comment (clintonb) [inline setup.py]: Do the same for all of the requirements, including those you did not set.

**Comment by clintonb** [comment]
Part of the reason you missed this in #18 is the fact that we are recording requirements in two places. This is wrong and error-prone. I recommend doing the following:

- Update `.travis.yml` to simply install `requirements.txt`
- Delete `test.txt` and `test_requirements.txt` since they are empty
- Remove all non-tests requirements from `requirements.txt` and replace them with `-e .` to install the package (and its requirements). ([This article](http://mike.zwobble.org/2013/05/adding-git-or-hg-or-svn-dependencies-in-setup-py/) should help with the `opaque_keys` requirement.)

This will ensure no one ever has an issue of forgetting requirements again.

File: requirements.txt [hunk 2]
```diff
@@ -1,12 +1,12 @@
 # Django/Framework Packages
 django>1.8,<1.9
 django-model-utils==1.4.0
+-e git+https://github.com/edx/django-rest-framework.git@3c72cb5ee5baebc4328947371195eae2077197b0#egg=djangorestframework==3.2.3
```
Comment (clintonb) [inline requirements.txt]: Why is this here? All of the non-test-related packages need to go away. The first line of this file should simply be: `-e .`

**Comment by clintonb** [comment]
@ahsan-ul-haq I made some changes. Take a look, and let me know your thoughts.

**Comment by ahsan-ul-haq** [comment]
@clintonb changes looks good to me but there were some requirements that were not included so I made some small changes as well. Let me know after review.

File: setup.py [hunk 3]
```diff
@@ -19,19 +22,13 @@
...
-    ],
+    packages=find_packages(exclude=['tests']),
     install_requires=[
-        "django>=1.8",
-        "django-model-utils==1.4.0",
+        'django>=1.8,<1.9',
+        'django-model-utils>=1.4.0,<1.5.0',
+        'djangorestframework>=3.2.0,<3.4.0',
+        'djangorestframework-jwt>=1.6.0,<=1.7.2',
+        'edx-opaque-keys>=0.1.2,<=0.2.0',
```
Comment (clintonb) [inline setup.py]: `<1.0.0` was correct here. We follow [semantic versioning](http://semver.org/), so all minor and patch changes should be compatible.

**Comment by clintonb** [comment]
:+1:

**Comment by zubair-arbi** [comment]
lgtm :+1:

**Comment by zubair-arbi** [comment]
nit: please bump the version in `setup.py`

**Comment by clintonb** [comment]
Reset the version, please. The version of 0.3.0 that you created earlier was incomplete. This is the real 0.3.0. Also, if this version were to be changed, it should have been 0.3.1 as there were no additions to the functionality.
